File: apps/cron/src/jobs/sessions.ts
import type { Database } from "@raven/db";
import { sessions } from "@raven/db";
import { lt } from "drizzle-orm";

export const cleanupExpiredSessions = async (db: Database) => {
  const now = new Date();

  const deleted = await db
    .delete(sessions)
    .where(lt(sessions.expiresAt, now))
    .returning({ id: sessions.id });

  console.log(`Session cleanup: deleted ${deleted.length} expired sessions`);
};

File: apps/cron/src/jobs/sessions.py
def cleanup_expired_sessions(conn) -> None:
    with conn.cursor() as cur:
        cur.execute("""
            DELETE FROM sessions
            WHERE "expiresAt" < NOW()
        """)
        count = cur.rowcount
        if count > 0:
            print(f"  Deleted {count} expired session(s)")
